File: src/main/java/jisop/command_line/parse/UnrecognizedArgument.java
package jisop.command_line.parse;

/**
 *
 * @author mathieu
 */
public class UnrecognizedArgument {
    public final int index;
    public final String value;
    public UnrecognizedArgument(int index, String value) {
        this.index = index;
        this.value = value;
    }
    @Override
    public int hashCode() {
        int hash = 3;
        hash = 41 * hash + this.index;
        hash = 41 * hash + (this.value != null ? this.value.hashCode() : 0);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final UnrecognizedArgument other = (UnrecognizedArgument) obj;
        return this.index == other.index &&
                this.value.equals(other.value);
    }

}

File: src/main/java/isop/command_line/parse/UnrecognizedArgument.kt
package isop.command_line.parse

/**
 *
 * @author mathieu
 */
class UnrecognizedArgument(val index: Int, val value: String?) {
    override fun hashCode(): Int {
        var hash = 3
        hash = 41 * hash + this.index
        hash = 41 * hash + if (this.value != null) this.value.hashCode() else 0
        return hash
    }

    override fun equals(obj: Any?): Boolean {
        if (obj == null) {
            return false
        }
        if (javaClass != obj.javaClass) {
            return false
        }
        val other = obj as UnrecognizedArgument?
        return this.index == other!!.index && this.value == other.value
    }

}
